src/agent/src/rpc.rs

```diff
@@ -1722,6 +1747,150 @@ fn load_kernel_module(module: &protocols::agent::KernelModule) -> Result<()> {
+    let target_path_oci = format!("{}{}", "oci://", IMAGE_OCI);
+
+    // Create directory into which to copy the manifest image
+    let status = Command::new("mkdir")
+        .arg("-v")
+        .arg("-p")
+        .arg(&IMAGE_MANIFEST)
+        .status()
+        .expect("Cannot create directory");
+
+    assert!(status.success());
+
+    // Create directory into which to copy the OCI image
+    // Will add code to munge the image registry/namespace/repository/tag to make this more dynamic
+    let status = Command::new("mkdir")
+        .arg("-v")
+        .arg("-p")
+        .arg(&IMAGE_OCI)
+        .status()
+        .expect("Cannot create directory");
+
+    assert!(status.success());
+
+    // TODO - fix up this nasty hack - skopeo won't accept blank credentials, so set them to something
+    let creds = if src_creds.trim().is_empty() {
+        "<blank>"
+    } else {
+        src_creds
+    };
+
+    let status = Command::new(SKOPEO_PATH)
+        .arg("copy")
+        .arg(source_image)
+        .arg(&target_path_manifest)
+        .arg("--src-creds")
+        .arg(creds)
+        .status()
+        .expect("Failed to pull image");
+
+    assert!(status.success());
+
+    // Copy image from one local file-system to another
+    // Resulting image is still stored in manifest format, but no longer includes the signature
+    // The image with a signature can then be unpacked into a bundle
+    let status = Command::new(SKOPEO_PATH)
+        .arg("copy")
+        .arg(&target_path_manifest)
+        .arg(&target_path_oci)
+        .arg("--remove-signatures")
+        .status()
+        .expect("Failed to copy image");
+
+    assert!(status.success());
+
+    Ok(())
+}
+
+#[allow(dead_code)]
+fn verify_image(image: &str, gpg_key: &str) -> Result<()> {
+    let target_path_manifest = IMAGE_MANIFEST;
+    let signature_file = format!("{}{}", target_path_manifest, "/signature-1");
+    let manifest_file = format!("{}{}", target_path_manifest, "/manifest.json");
+
+    // Create a directory into which to import the public key
+    let status = Command::new("mkdir")
+        .arg("-p")
+        .arg(GPG_HOME)
+        .status()
+        .expect("Cannot create directory");
+
+    assert!(status.success());
+
+    // Import the public key
+    let status = Command::new("gpg")
+        .arg("--import")
+        .arg("/root/davehay_pub.gpg")
+        .status()
+        .expect("Cannot import public key");
+
+    assert!(status.success());
+
+    // Verify image
+    let status = Command::new(SKOPEO_PATH)
+        .arg("standalone-verify")
+        .arg(manifest_file)
+        .arg(image)
+        .arg(gpg_key)
+        .arg(signature_file)
+        .status()
+        .expect("Failed to verify signature");
+
+    assert!(status.success());
+
+    Ok(())
+}
+
+fn unpack_image(cid: &str) -> Result<()> {
+    // To save space delete the manifest. This was only needed as long as verify image
+    let status = Command::new("rm")
```

Comment:
Here as well, we should use https://doc.rust-lang.org/std/fs/fn.remove_dir_all.html instead.